forum/api.py

```diff
@@ -0,0 +1,34 @@
+"""
+Native Python APIs.
+"""
+
+from typing import Any
+
+from forum.api_utils import (
+    pin_unpin_thread,
+    retrieve_user_data,
+)
+from forum.models.model_utils import get_commentables_counts_based_on_type
+
+
+def pin_thread(user_id: str, thread_id: str) -> dict[str, Any]:
+    """Pin a thread."""
+    return pin_unpin_thread(user_id, thread_id, "pin")
```

Comment:
That dict[str, Any] return value is very confusing. What are the keys and values supposed to be? At the very least, the API methods should be documented to describe the return values. Alternatively, depending on the content of the return value, we should return more explicit arguments. Once we ditch the REST API, we no longer have a good reason to return dict objects.

Reply:
hmmm, but we are sending whole thread's data as a return value which is a dict and that's what edx-platform is expecting from this native API.

Reply:
we can do same or like this as we were doing for HTTPs APIs. e.g.
```
"""
Pins a thread.
Parameters:
    request (Request): The incoming request.
    thread_id (str): The ID of the thread to pin.
Body:
    user_id: requesting user's id
Response:
    A response with the updated thread data.
"""
```